cfme/tests/configure/test_workers.py

```diff
@@ -1,12 +1,23 @@
 # -*- coding: utf-8 -*-
 
 import pytest
-from cfme.configure import configuration
+
+from cfme.utils.wait import wait_for
 
 
 @pytest.mark.tier(3)
 @pytest.mark.sauce
 @pytest.mark.nondestructive
-def test_restart_workers():
-    assert configuration.restart_workers("Generic Worker", wait_time_min=3),\
-        "Could not correctly restart workers!"
+def test_restart_workers(appliance):
+    worker = appliance.collections.diagnostic_workers.instantiate(name="Generic Worker")
+    pids = worker.reload_worker()
+    # Wait for all original workers to be gone
+    wait_for(worker.check_workers_finished, func_args=[pids],
+             fail_func=worker.parent.reload_workers_page,
+             num_sec=1800, delay=10, message="Wait for all original workers to be gone"
+             )
```

Comment:
**Optional**
Bump these trailing parenthesis up into the previous line if they fit in 100 chars